lib/Migration/Version0104Date20200314074611.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * @copyright Copyright (c) 2017 René Gieling <github@dartcafe.de>
+ *
+ * @author René Gieling <github@dartcafe.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Polls\Migration;
+
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\Migration\SimpleMigrationStep;
+use OCP\Migration\IOutput;
+
+/**
+ * Installation class for the polls app.
+ * Initial db creation
+ */
+class Version0104Date20200314074611 extends SimpleMigrationStep {
+
+	/** @var IDBConnection */
+	protected $connection;
+
+	/** @var IConfig */
+	protected $config;
+
+	/**
+	 * @param IDBConnection $connection
+	 * @param IConfig $config
+	 */
+	public function __construct(IDBConnection $connection, IConfig $config) {
+		$this->connection = $connection;
+		$this->config = $config;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param \Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 * @since 13.0.0
+	 */
+	public function changeSchema(IOutput $output, \Closure $schemaClosure, array $options) {
+		// get table prefix, as we are running a raw query
+        $prefix = $this->config->getSystemValue('dbtableprefix', 'oc_');
+        // check for orphaned entries in all tables referencing
+        // the main polls table
+        $query = "ALTER TABLE {$prefix}polls_polls
+				CHANGE description description TEXT;";
```

Comment:
Hi @dartcafe,

Can you use Doctrine DSL instead of raw SQL please?
Because this query doesn't seem to be compatible with PostgreSQL and SQLite.

Have a good day,
Tortue Torche

Reply:
F****ing doctrine. Took several approches and pattern searching to get this done without adding and dropping a column and fiddeling with temporary columns. 

Does anybody know a good and understandable documentation of this? FInally found the hints of how to use `changeColumn`.